client/src/component/itemList.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from 'react';
+import List from '@material-ui/core/List';
+import Button from '@material-ui/core/Button';
+import { v4 as uuidv4 } from 'uuid';
+import Item from './item';
+
+const ItemList = ({setTicketItems}) => {
+    const [itemList, setItemList] = useState([]);
+
+    const removeItem = (index) =>{
+        const newItemList = itemList.filter(item => item.id !== index);
+        setItemList(newItemList);
+    }
+
+    useEffect(() => {setTicketItems(itemList)}, [itemList])
+
+    const updateItemList = (index, value, _prop) => {
+        const newItemList = itemList.map((item) => {
+            if(item.id === index)
+            {
+                item[_prop] = value;
+            }
+            return item;
+        });
+        setItemList(newItemList)
+    }
+
+    const addItem = () => { 
+        let newId = uuidv4();
```

Comment:
I like this library overall, but it's probably best to let the server set your IDs.

Reply:
Hey Zac, the id I am setting here is temporary and it's being used to manage the state changes/deletion for the item component. When I end up saving the ticket to the DB I do not pass the id.

Reply:
In that case, maybe calling `newId` `tempId` instead will avoid this confusion.

Reply:
Yup, sounds good. I made the change.